src/definitions/list.ts

```diff
@@ -0,0 +1,60 @@
+import { GetGen } from '../typegenTypeHelpers'
+import { AllNexusNamedTypeDefs } from './wrapping'
+import { NexusTypes, withNexusSymbol } from './_types'
+
+type AllNamedTypes = GetGen<'allNamedTypes', string> | AllNexusNamedTypeDefs
+
+/**
+ * list()
+ */
+
+type ListableTypes = AllNamedTypes | NexusNonNullDef<NonNullableTypes> | NexusListDef<ListableTypes>
+
+export type NexusListDefConfig<TypeName extends ListableTypes> = {
+  type: TypeName
+}
+
+export class NexusListDef<TypeName extends ListableTypes> {
+  // @ts-ignore
+  private _isNexusListDef: boolean = true
+
+  constructor(readonly type: TypeName) {}
+}
+
+withNexusSymbol(NexusListDef, NexusTypes.List)
+
+export function list<TypeName extends ListableTypes>(type: TypeName) {
+  return new NexusListDef(type)
+}
+
+/**
+ * nonNull()
+ */
+
+type NonNullableTypes = AllNamedTypes | NexusListDef<ListableTypes>
```

Comment:
I wonder if we should just add `NexusNonNullDef<NonNullableTypes>` to this, so you can wrap `nonNull(nonNull(...))` and have it be a no-op. IIRC that's how GraphQL's definition of non-nullable works

Reply:
Ohhh, I didn't know about that! Yeah, that would also simplify the types 👍

Reply:
Turns out I was completely wrong about this. Don't know what I thought I was remembering. Thanks to @Weakky for pointing this out 🤦